### PR TITLE
Add simple test helper, extracted from janet-lang

### PR DESCRIPTION
**Pull request by pepe**
I am using this helper, extracted from other janet-lang libraries, in some of my libs, usually with spork already there, so I think it could be a good idea to have it here.

I am also thinking about moving `timeit` fn from `misc` and extract `capture-stdin` into this module. What do you think, @MikeBeller & @bakpakin?

### CODE REVIEW

File: spork/test/init.janet [hunk 1]
```diff
@@ -25,21 +25,25 @@
   x)
 
 (defmacro assert-error
+  "Test passes if forms error."
   [msg & forms]
   (def errsym (keyword (gensym)))
   ~(,assert (= ,errsym (try (do ,;forms) ([_] ,errsym))) ,msg))
 
 (defmacro assert-no-error
+  "Test passes if forms not error."
```
Comment (sogaiu) [inline spork/test/init.janet:34]: How about:

> Test passes if forms do not error.

?

Comment (pepe) [reply spork/test/init.janet]: yep

**Comment by pepe** [comment]
Funny thing, I have not added tests for this module. I will do it later today.

**Comment by MikeBeller** [comment]
Yes I think adding the very basic testing capability to spork is a very good idea.  And if you do that it makes sense to move timeit in there, and add capture-stdout.  I really didn't test capture-stdout well since it was a one-off inside a test suite so if you are going to add it maybe add a few tests for it too?

**Comment by pepe** [comment]
Yep, I will add tests for the whole test module.

**Comment by pepe** [comment]
So I added tests for `spork/test` and made all the other tests use `spork/test`, which reminded me about Uroboros.

I also removed the old `test/helper.janet` after extracting the `suppress-stdout` macro from it.

**Comment by bakpakin** [comment]
Makes sense, that bit of code was being copied around a lot...

LGTM.

**Pull request merged**